src/services/authServices.ts

```diff
@@ -0,0 +1,35 @@
+import prisma from '../utils/client';
+import bcrypt from 'bcrypt';
+
+interface UserData {
```

Comment:
Is this the same type as TRegisterRequest? Maybe I didnt notice in the first go, but type inferred by zod schema should be used here as well.